schema/crdb/dbinit.sql

```diff
@@ -2534,6 +2534,31 @@ CREATE TABLE IF NOT EXISTS omicron.public.tuf_repo_artifact (
     PRIMARY KEY (tuf_repo_id, tuf_artifact_id)
 );
 
+-- Describes an individual association between an RoT or RoT bootloader artifact
+-- and its sign (RKTH) value from an uploaded TUF repo.
+CREATE TABLE IF NOT EXISTS omicron.public.tuf_rot_by_sign (
+    id UUID PRIMARY KEY,
+    name STRING(63) NOT NULL,
+    version STRING(63) NOT NULL,
+    -- The kind of artifact this is
+    kind STRING(63) NOT NULL,
+    -- The sign (RKTH) value associated with this image
+    sign Bytes NOT NUll,
+    -- The time this rot_by_sign was first recorded
+    time_created TIMESTAMPTZ NOT NULL,
+    CONSTRAINT unique_name_version_kind_sign UNIQUE (name, version, kind, sign)
+);
+
+-- Reflects that a particular RoT or RoT bootloader artifact was provided by a
+-- particular TUF repo.
+-- This is a many-many mapping.
+CREATE TABLE IF NOT EXISTS omicron.public.tuf_repo_rot_by_sign (
+    tuf_repo_id UUID NOT NULL,
+    tuf_rot_by_sign_id UUID NOT NULL,
+
+    PRIMARY KEY (tuf_repo_id, tuf_rot_by_sign_id)
+);
```

Comment:
It feels somewhat dubious to have a separate table with _separate_ "ROT by sign IDs" (which are presumably different from the IDs in the `tuf_artifact` table) that is describing optional data on a TUF artifact. Given the UNIQUE constraint I'm tempted to suggest that this should be a nullable column on the main `tuf_artifact` table.